proto/viam/app/data/v1/data.proto

```diff
@@ -124,3 +133,33 @@ message BinaryMetadata {
   string file_ext = 6;
   string uri = 7;
 }
+
+// DeleteTabularDataByFilterRequest deletes the data and metadata of tabular data when a filter is provided
+message DeleteTabularDataByFilterRequest {
+  Filter filter = 1;
+}
+
+// DeleteBinaryDataByFilterResponse returns the number of tabular datapoints deleted when a filter is provided
+message DeleteTabularDataByFilterResponse {
+  int64 deleted_count = 1;
+}
+
+// DeleteBinaryDataByFilterRequest deletes the data and metadata of binary (image + file) data when a filter is provided
+message DeleteBinaryDataByFilterRequest {
+  Filter filter = 1;
+}
+
+// DeleteBinaryDataByFilterResponse returns the number of binary files deleted when a filter is provided
+message DeleteBinaryDataByFilterResponse {
+  int64 deleted_count = 1;
+}
+
+// DeleteBinaryDataByIDsRequest deletes the data and metadata of binary (image + file) data by file ids
```

Comment:
super nit: "... data and metadata of binary (image + file) data _when file ids are provided_"

Reply:
Thanks, done!